pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -62,10 +64,10 @@ type Values struct {
 	Name string
 	// SecretName is the name of the secret referenced by the DNSRecord resource.
 	SecretName string
-	// ReconcileOnce specifies that the DNSRecord resource should only be created and never reconciled after that again.
-	// However, on Deploy it is still updated with the timestamp annotation.
-	// This mode is used for owner DNS records.
-	ReconcileOnce bool
+	// ReconcileOnValueChangeOnly specifies that the DNSRecord resource should only be created and only reconciled when Values are changed.
+	// If Values didn't change, it is still updated with the timestamp annotation on Deploy.
+	// This mode is used for owner DNS records to avoid competing reconciliations during controlplane migrations.
+	ReconcileOnValueChangeOnly bool
```

Comment:
I would really prefer a shorter name such as `ReconcileOnChange` (open to suggestions but ideally no longer than 3 words). Also, the current name is inaccurate - we want to reconcile when the desired state has changed compared to the current one. The `Values` struct is an implementation detail and it's always newly created (so it can't really "change").
```suggestion
	// ReconcileOnChange specifies that the DNSRecord resource should only be reconciled when first created or if its desired state has changed compared to the current one.
	// This mode is used for owner DNS records to avoid competing reconciliations during the control plane migration "bad case" scenario.
	ReconcileOnChange bool
```

Reply:
done.